### PR TITLE
feat: support `load_mode` and `thread` option for LOAD DATA

### PR DESCRIPTION
**Pull request by zhanghaohit**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
when `LOAD DATA` in cluster mode, we provide options to allow load locally without `task manager`/`spark`.

### CODE REVIEW

File: src/sdk/sql_cluster_router.cc [hunk 1]
```diff
@@ -2947,19 +2956,71 @@ ::openmldb::base::Status SQLClusterRouter::SaveResultSet(const std::string& file
...
+    }
+    return status;
+}
+
+hybridse::sdk::Status SQLClusterRouter::LoadDataMultipleFile(int id, int step, const std::string& database,
+                                                             const std::string& table,
+                                                             const std::vector<std::string>& file_list,
+                                                             const openmldb::sdk::ReadFileOptionsParser& options_parser,
+                                                             uint64_t* count) {
+
```
Comment (zhanghaohit) [inline src/sdk/sql_cluster_router.cc]: lint

**Comment by zhanghaohit** [comment]
add docs

File: src/sdk/file_option_parser.h [hunk 2]
```diff
@@ -32,11 +32,13 @@ namespace sdk {
 class FileOptionsParser {
  public:
     FileOptionsParser() {
+        quote_ = '\0';
         check_map_.emplace("format", std::make_pair(CheckFormat(), hybridse::node::kVarchar));
         check_map_.emplace("delimiter", std::make_pair(CheckDelimiter(), hybridse::node::kVarchar));
         check_map_.emplace("null_value", std::make_pair(CheckNullValue(), hybridse::node::kVarchar));
         check_map_.emplace("header", std::make_pair(CheckHeader(), hybridse::node::kBool));
         check_map_.emplace("quote", std::make_pair(CheckQuote(), hybridse::node::kVarchar));
+        check_map_.emplace("mode", std::make_pair(CheckMode(), hybridse::node::kVarchar));
```
Comment (zhanghaohit) [inline src/sdk/file_option_parser.h:41]: - add mode checking for online load

File: src/base/file_util.h [hunk 3]
```diff
@@ -328,6 +344,56 @@ inline static int HardLinkDir(const std::string& src, const std::string& dest) {
     return 0;
 }
 
+// list of paths of all files under the directory 'dir' when the filename matches the pattern
+// do not include any subfolder
+inline static std::vector<std::string> FindFiles(const std::string& dir, const std::string& pattern) {
+    // convert ls pattern `test*` to regex pattern `test.*`
+    std::string converted_pattern;
+    for (auto c : pattern) {
```
Comment (zhanghaohit) [inline src/base/file_util.h]: `glob`

Comment (aceforeverd) [reply src/base/file_util.h]: c glob seems not available for windows.

Comment (zhanghaohit) [reply src/base/file_util.h]: yes. it is POSIX. anyway most of methods in file_util.h is linux-specific.

File: src/base/file_util.h [hunk 4]
```diff
@@ -17,14 +17,19 @@
...
 #include <dirent.h>
+#include <glob.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <string.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
 
+#include <regex>
```
Comment (dl239) [inline src/base/file_util.h]: rm unused header

**Comment by dl239** [comment]
> add docs

docs will be add in this pr later?

**Comment by zhanghaohit** [comment]
> > add docs
> 
> docs will be add in this pr later?

yes. done the docs

**Review by dl239 — APPROVED** [review]

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**